### PR TITLE
feat: Fix Calendar updateEvent bug + Gmail enhancements (v3.3.1)

### LINKED ISSUE #31: 🐛 Bug: updateEvent operation fails with 'Cannot read properties of undefined (reading 'start')'
## Bug Description

The `updateEvent` operation in the calendar module fails with the error:
```
MCP error -32603: Cannot read properties of undefined (reading 'start')
```

This occurs regardless of how the start/end times are formatted in the parameters.

## Steps to Reproduce

1. Create a calendar event using `quickAdd`:
```json
{
  "operation": "quickAdd",
  "params": {
    "text": "Meeting January 10 2pm-3pm"
  }
}
```

2. Attempt to update the event to add attendees:
```json
{
  "operation": "updateEvent",
  "params": {
    "eventId": "<event-id>",
    "summary": "Updated Title",
    "description": "Meeting description",
    "start": "2026-01-10T14:00:00-06:00",
    "end": "2026-01-10T15:00:00-06:00",
    "attendees": ["user@example.com"],
    "sendUpdates": "all"
  }
}
```

## Expected Behavior

The event should be updated with the new title, description, and attendees. Invites should be sent to the specified attendees.

## Actual Behavior

The operation fails with:
```
MCP error -32603: Cannot read properties of undefined (reading 'start')
```

## Variations Attempted

All of these parameter formats failed with the same error:

### Format 1: ISO string
```json
{
  "start": "2026-01-10T14:00:00-06:00",
  "end": "2026-01-10T15:00:00-06:00"
}
```

### Format 2: Object with dateTime
```json
{
  "start": {"dateTime": "2026-01-10T14:00:00-06:00", "timeZone": "America/Chicago"},
  "end": {"dateTime": "2026-01-10T15:00:00-06:00", "timeZone": "America/Chicago"}
}
```

### Format 3: startTime/endTime
```json
{
  "startTime": "2026-01-10T14:00:00-06:00",
  "endTime": "2026-01-10T15:00:00-06:00"
}
```

## Root Cause Hypothesis

The `updateEvent` handler likely expects a specific parameter structure but is receiving `undefined` when accessing `params.start` or a nested property. The validation/parsing logic may not be handling the input correctly.

## Acceptance Criteria

- [ ] `updateEvent` operation successfully updates calendar events
- [ ] Attendees can be added to existing events
- [ ] `sendUpdates` parameter sends invites to attendees
- [ ] Support both ISO string and object formats for start/end times
- [ ] Add input validation with clear error messages for malformed parameters

## Technical Notes

**File to investigate:** `src/tools/calendar.ts` (or similar)

**Related operations that work:**
- `quickAdd` - works correctly
- `createEvent` - also has issues (different error)
- `deleteEvent` - works correctly
- `getEvent` - works correctly

## Priority

P2 - Medium (workaround exists: send email invites separately)

### PR DESCRIPTION
**Pull request by AojdevStudio**
## Summary

- **Fix Calendar updateEvent bug (#31)** - Now accepts both ISO strings and EventDateTime objects for start/end times
- **Add Gmail updateDraft operation** - Update existing drafts while preserving unchanged fields
- **Add Gmail attachment operations** - listAttachments and getAttachment for email attachment handling
- **Archive legacy handlers** - Move deprecated code to archive/legacy-handlers-v2/

### Calendar Module (Bug Fix)
- Add `normalizeEventDateTime()` utility function to convert string/object inputs
- Add `FlexibleDateTime` type for flexible datetime input
- Update `UpdateEventOptions` to accept both formats
- Add 23 unit tests for datetime normalization
- Clear error messages with format examples

### Gmail Module (Enhancements)
- `updateDraft` - Update draft content (preserves fields not provided)
- `listAttachments` - List attachment metadata from a message
- `getAttachment` - Download attachment content as base64

### Tech Debt
- Archive legacy handlers (src/drive, src/sheets, src/forms, src/docs)
- Archive legacy tests that depended on archived code
- Update jest.config.js to ignore archive directory

## Test plan

- [x] Type checking passes (`npx tsc --noEmit`)
- [x] All tests pass (322/322) - `npm test`
- [x] Calendar utils tests (23 new tests)
- [x] No breaking changes to existing operations
- [ ] Manual: Test updateEvent with ISO string times in Claude Desktop
- [ ] Manual: Test Gmail updateDraft flow

## Files Changed

| Area | Files |
|------|-------|
| Calendar | `src/modules/calendar/{utils,types,update,index}.ts` |
| Gmail | `src/modules/gmail/{compose,types,attachments,index}.ts` |
| Main | `index.ts` (dispatch), `src/tools/listTools.ts` |
| Tests | `src/modules/calendar/__tests__/utils.test.ts` |
| Config | `jest.config.js` |
| Archive | `archive/legacy-handlers-v2/*` |

Closes #31

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Gmail: update drafts, list attachments, and download attachments.
  * Calendar: updateEvent accepts flexible datetime formats (ISO strings or event objects).
  * RBP tooling: new CLI scripts for spec/story import, autonomous execution loop, test-gated closure, sequencing, and observability/event emission.

* **Documentation**
  * Extensive docs and quick-starts for project, RBP workflows, command usage, and onboarding.

* **Tests**
  * Added unit tests for calendar datetime normalization and edge cases.

* **Chores**
  * Added project ignore rules and attributes to avoid committing runtime/state/merge artifacts.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .claude/commands/rbp/start.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+   ```bash
+   ~/.claude/skills/Observability/manage.sh start
+   ```
+   - Wait up to 10 seconds for startup
+   - Verify with health check
+
+4. **Open browser (unless headless):**
+   - Check for CI/headless environment variables: `$CI`, `$GITHUB_ACTIONS`, `$GITLAB_CI`, `$JENKINS_URL`, `$CODESPACES`
+   - Check for SSH without display: `$SSH_CONNECTION` without `$DISPLAY`
+   - If not headless: Open http://localhost:5172 in browser
```
[the remark this answers is not in the document]
Comment (AojdevStudio) [reply .claude/commands/rbp/start.md]: This is not actual product code ignore

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+# Use bd merge for beads JSONL files
+.beads/issues.jsonl merge=beads
```
[the remark this answers is not in the document]
Comment (AojdevStudio) [reply .gitattributes]: ignore
